### ISSUE TITLE
Breaks if using Nix string interpolation inside the Lua code in Wezterm config

### ISSUE THREAD

**Issue by mrjones2014** [calops/hmts.nvim#15]
On line 44, I do string interpolation, and the Lua highlighting is broken after that.

![CleanShot 2023-10-30 at 07 31 51](https://github.com/calops/hmts.nvim/assets/8648891/8a0fc1fb-bf39-421b-99f3-7c5fe2acdd46)

**Comment by calops** [comment]
Hi, thanks for the report!

Unfortunately, I don't think this is solvable right now (if ever). To handle interpolation, we have two choices:

- Highlight the whole `indented_string_expression` as the injected language. Problem: this includes the interpolation pattern in the language, which more often than not will not match the language's syntax and mess the highlighting in random ways.
- Highlight every `string_fragment` as the injected language. This is the currently implemented solution. Problem: all the fragments are independent from each other, which means a rogue quote character at the beginning will mess up the highlighting.

I don't see a third option that would have neither drawback.

**Comment by mrjones2014** [comment]
Ah, that is unfortunate. In the mean time, I can get around this formatting the interpolation differently -- putting the Lua quotes inside the string outputted by Nix e.g. instead of `local os_name = '${if isLinux then "linux" else "macos"}'`I can do `local os_name = ${if isLinux then "'linux'" else "'macos'"}` and it works.

**Comment by calops** [comment]
Yeah that's the workaround I'm using, but I agree that it's not satisfying. Unfortunately that's an understandable limitation of treesitter itself.
